### ISSUE TITLE
Add a sbo__relationship widget shortcut for "self"

### ISSUE THREAD

**Issue by dsenalik** [tripal/tripal#1326]
## Feature Request/Discussion
* This feature does attempt to solve an existing problem with Tripal
* I will implement this if it is a good idea.
* This feature is Not Urgent

### Description
When entering a relationship with the sbo__relationship field widget, it is required that either the subject or the object refers to the current record. To avoid copy+paste the entity name, or use the ajax autocomplete which in some cases returns a lot of records, I propose a mechanism to easily select "this" or "self" on either of the two fields. 

But how should it work? Discuss.

1. Type "self" or "this" or "." or some other secret word in the box. Maybe the ajax callback could be aware and then expand it. But what if records start with this secret word?
2. An empty box means self. (Probably easiest to implement and no change to user interface)
3. Add a radio button to each box to select which one is the current record.
4. Add a check box to the front "Subject refers to this record" and only have one box for input (plus cv).
5. Something else?

**Comment by laceysanderson** [comment]
At one point we had the radio button approach but it made the interface too cluttered. The empty box referring to self used to work (which is what I was trying to mention in the review comment -that I noticed it's broken now) although apparently it needs better in-form documentation so people know it's there 🙈

**Comment by laceysanderson** [comment]
It turns out that adding relationships during Tripal content creation was broken in general in 7.x-3.x right now so this is a bug that we will want to fix.

In 7.x-3.x right now when you try this you get an ajax error as follows:
```
An AJAX HTTP error occurred.
HTTP Result Code: 500
Debugging information follows.
Path: /system/ajax
StatusText: Service unavailable (with message)
ResponseText: TypeError: count(): Argument #1 ($value) must be of type Countable|array, DatabaseStatementBase given in sbo__relationship-&gt;validateItem() (line 1107 of /var/www/html/sites/all/modules/contrib/tripal/tripal_chado/includes/TripalFields/sbo__relationship/sbo__relationship.inc).
```

The expected behaviour should be:
When creating a new piece of content named organism2, with a relationship of organism1 (subject) synonym_type:related (type) organism2 (object) this is what you would expect to be able to do:
1. You set the subject to organism1 which must already exist and thus will come up in the drop-down.
2. You select synonym_type (the vocabulary containing the type of relationship you are adding) and ajax is triggered with no noticable change (it's setting up the type textfield autocomplete).
3. You enter the name of the type of relationship: related.
4. You leave the object empty since it is the current content being created and thus can't be looked up with the autocomplete.
5. You click save/create and everything validates successfully and the relationship mentioned above is added.

This was working at some point. The storage system will automatically fill that in if left empty -this is working in properties for example which can be created on content creation despite the base record not existing yet and the chado primary key thus being empty. So if we can fix the validation for this field then we should be golden.

**Comment by dsenalik** [comment]
I'm pretty sure that ajax error is from the ```chado_linker__contact``` field 🤔, you can see an isolated fix at https://github.com/tripal/tripal/pull/1267/commits/5949490bdd9eb03e2a3895fe80c12e2d798469d4

No I was wrong, I should have read the whole error message, it was the missing ```->fetchAll()``` at the referenced line (which was not giving an error in PHP7).

**Comment by dsenalik** [comment]
This is mostly implemented with https://github.com/tripal/tripal/pull/1267/commits/24b8c40d408cac302978d77cb1043417e0473528 on pull request #1267 
where "an empty box means self" is revived into functionality, thanks @laceysanderson for the help along the way.

**Comment by dsenalik** [comment]
Fully implemented with merged #1267 and #1396
